nym-node/nym-node-http-api/src/router/api/v1/gateway/client_interfaces/wireguard/client_registry.rs

```diff
@@ -11,33 +11,59 @@ use axum::extract::{Path, Query, State};
 use axum::http::StatusCode;
 use axum::Json;
 use nym_node_requests::api::v1::gateway::client_interfaces::wireguard::models::{
-    ClientMessage, ClientRegistrationResponse, GatewayClient, InitMessage, Nonce, PeerPublicKey,
+    ClientMessage, ClientRegistrationResponse, GatewayClient, InitMessage, PeerPublicKey,
 };
+use nym_wireguard_types::registration::RegistrationData;
 use rand::{prelude::IteratorRandom, thread_rng};
 
+fn remove_from_registry(
+    state: &WireguardAppStateInner,
+    remote_public: &PeerPublicKey,
+    gateway_client: &GatewayClient,
+) -> Result<(), RequestError> {
+    state
+        .wireguard_gateway_data
+        .remove_peer(gateway_client)
+        .map_err(|err| RequestError::from_err(err, StatusCode::INTERNAL_SERVER_ERROR))?;
```

Comment:
are we sure we want to be returning the error immediately here? because if we fail to remove the peer for some weird reason, the public key will still be stuck in the `client_registry`

Reply:
Once the `client_registry` will be made persistent, we will have an error path there as well, so we'd need to decide which one we regard as a source of truth, so that it can be the first operation. Right now, my thinking is that, since kernel data is the only one that can fail, if it's not removed from that list, there's no reason to remove it from the registry either, in case the weird reason disappears later.
tl;dr I agree the removal error case needs to be carefully ordered here, but for now, as `client_registry` doesn't do much, it's not so important.